### PR TITLE
Refactor macros

### LINKED ISSUE #55: simplify histogram register macro
- use Vec<f64> or &[f64] for buckets in register histogram macro
- add bucket arg for register histogram vec macro

### PR DESCRIPTION
**Pull request by overvenus**
Macro API changes:

* `register_(counter|gauge)`
  * Reomved:
    - `( $ NAME : expr , $ HELP : expr $ ( , $ CONST_LABELS : expr ) * )`

* `register_(counter_vec|gauge_vec)`
  * Reomved
    - `( $ NAME : expr , $ HELP : expr , $ CONST_LABELS : expr , $ LABELS_NAMES : expr )`

* `histogram_opts!`
  * Remove: 
    - `histogram_opts!( $ NAME : expr , $ HELP : expr , [ $ ( $ BUCKETS : expr ) , * ] )`
    - `histogram_opts!( $ NAME : expr , $ HELP : expr $ ( , $ CONST_LABELS : expr ) * )`
    - `histogram_opts!( $ NAME : expr , $ HELP : expr , $ CONST_LABELS : expr , [ $ ( $ BUCKETS : expr ) , + ] )`
  * Add:
    - ~~`histogram_opts!( $ OPTS : expr , $ BUCKETS : expr )`~~
    - `histogram_opts!( $ NAME : expr , $ HELP : expr )`
    - `histogram_opts!( $ NAME : expr , $ HELP : expr , $ BUCKETS : expr )`
    - `histogram_opts!( $ NAME : expr , $ HELP : expr , $ BUCKETS : expr , $ CONST_LABELS : expr )`

* `register_histogram!`
  * Remove: 
    - `register_histogram!( $ NAME : expr , $ HELP : expr , $ CONST_LABELS : expr )`
    - `register_histogram!( $ NAME : expr , $ HELP : expr , $ CONST_LABELS : expr , [ $ ( $ BUCKETS : expr ) , + ] )`
  * Add:
    - `register_histogram!( $ NAME : expr , $ HELP : expr , $ BUCKETS : expr )`

* `register_histogram_vec!`
  * Remove: 
    - `register_histogram_vec!( $ NAME : expr , $ HELP : expr , $ CONST_LABELS : expr , $ LABELS_NAMES : expr )`
  * Add:
    - `register_histogram_vec!( $ NAME : expr , $ HELP : expr , $ LABELS_NAMES : expr , $ BUCKETS : expr)`

In a nutshell, `CONST_LABELS` are removed from `register_(counter|counter_vec|gauge|gauge_vec|histogram|histogram_vec)`.

The only way to include `CONST_LABELS` is through `opts!`.

---------

~~Note that `histogram_opts!("name", "help")` is no longer available, it should be replaced with `histogram_opts!(opt!("name", "help"))`.~~

Close #55 and #90

PTAL @siddontang

### CODE REVIEW

File: src/macros.rs [hunk 1]
```diff
@@ -107,80 +107,44 @@ macro_rules! opts {
...
-///                             exponential_buckets(4.0, 1.1, 4).unwrap()]);
-/// assert_eq!(opts.buckets.len(), 8);
+/// let opts = histogram_opts!(name, help, linear_buckets(1.0, 0.5, 4).unwrap());
+/// assert_eq!(opts.buckets.len(), 4);
 /// # }
 /// ```
 #[macro_export]
 macro_rules! histogram_opts {
-    ( $ NAME : expr , $ HELP : expr , [ $ ( $ BUCKETS : expr ) , * ] ) => {
+    ( $ OPTS : expr ) => {
```
Comment (siddontang) [inline src/macros.rs]: I think following may be better. 
```
histogram_opts(name, help)
histogram_opts!(name, help, buckets)
histogram_opts!(name, help, buckets, const_labels)

```

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @kamalmarhubi

**Comment by siddontang** [comment]
> Note that histogram_opts!("name", "help") is no longer available, it should be replaced with histogram_opts!(opt!("name", "help")).

@overvenus you should update this.

**Comment by kamalmarhubi** [comment]
@siddontang anything in particular to look at?

**Comment by siddontang** [comment]
@kamalmarhubi 

Just to see whether the refactor macro is easy to use. :-)

**Comment by disksing** [comment]
LGTM
